server/setup.go

```diff
@@ -25,66 +36,92 @@ func NewState() *State {
 	//
 	// curl http://localhost:3000
 	r := chi.NewRouter()
-	// add logger for all requests
+
+	// add prebuild middleware for all requests
 	r.Use(middleware.Logger)
+	r.Use(middleware.RequestID)
+	r.Use(middleware.RealIP)
+	r.Use(middleware.Recoverer)
+
+	// add expvars at /debug/vars
+	expvar.Publish("counter200Code", &counter200Code)
+	expvar.Publish("counter400Code", &counter400Code)
+	expvar.Publish("counter500Code", &counter500Code)
+	// add pprof at /debug/
+	r.Mount("/debug", middleware.Profiler())
+
+	// add expvar and db metrics to prometheus
+	reg := prometheus.NewRegistry()
+	reg.MustRegister(collectors.NewBuildInfoCollector())
+	reg.MustRegister(collectors.NewDBStatsCollector(db.GetSqlDB(), "postgres"))
+	// TODO: figure out how to add types
+	reg.MustRegister(collectors.NewExpvarCollector(
+		map[string]*prometheus.Desc{
+			"counter200Code": prometheus.NewDesc("expvar_200Status", "number of status 200 api calls", nil, nil),
+			"counter400Code": prometheus.NewDesc("expvar_400status", "number of status 400 api calls", nil, nil),
+			"counter500Code": prometheus.NewDesc("expvar_500status", "number of status 500 api calls", nil, nil),
+		},
+	))
+	// add prometheus endpoint at /metrics. The above collectors will be show
+	// in the reverse order the are registered.
+	r.Mount("/metrics", promhttp.HandlerFor(
+		reg,
+		promhttp.HandlerOpts{
+			// Opt into OpenMetrics to support exemplars.
+			EnableOpenMetrics: true,
+		},
+	))
 
-	// TODO(soypete): investigate middleware to see if we should add any for intro
-	// topics or wait until later
-	// r.Use(middleware.RequestID)
-	// r.Use(middleware.RealIP)
-	// r.Use(middleware.Recoverer)
-	r.Get("/", func(w http.ResponseWriter, r *http.Request) {
-		w.Write([]byte("welcome to game server"))
-	})
-	r.Get("/health", func(w http.ResponseWriter, r *http.Request) {
-		w.Write([]byte("OK"))
-	})
-	// TODO(soypete): investigate middleware to see if we should add any for intro
-	// topics or wait until later
-	//r.Use(middleware.RequestID)
-	//r.Use(middleware.RealIP)
-	//r.Use(middleware.Recoverer)
 	r.Get("/", func(w http.ResponseWriter, r *http.Request) {
 		w.Write([]byte("welcome to game server"))
 	})
-	r.Get("/health", func(w http.ResponseWriter, r *http.Request) {
-		w.Write([]byte("OK"))
-	})
 
 	s := &State{
 		db:      db,
 		Router:  r,
 		BaseURL: "http://localhost:3000", // TODO: this should be a config
-		Port:    "3000",
+		Port:    ":3000",
 	}
 
+	// TODO: Change register routes.
+	// /register?username=...&password... (no auth)
+	// user/{username}/get (with auth)
+	// user/{username}/update?password=... (with auth)
+	// user/{username}/delete (with auth)
+
```

Comment:
These endpoints do not currently have auth on them. This means that the tests that validate auth are failing. The subsequent pr will include the user API refactor and fix those test